src/test/java/jenkins/plugins/office365connector/FormUtilsTest.java

```diff
@@ -0,0 +1,50 @@
+package jenkins.plugins.office365connector;
+
+import hudson.util.FormValidation;
+import jenkins.plugins.office365connector.util.FormUtils;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+
+public final class FormUtilsTest {
+
+    private static final String NOT_VALID = "demo";
+    private static final String NOT_VALID_URL = "hpp://demo";
+    private static final String NOT_VALID_VARIABLE_REFERENCE = "$$$$demo";
+    private static final String VALID_VARIABLE_REFERENCE = "$demo";
+    private static final String VALID_URL = "http://demo";
```

Comment:
looks like all above can be moved to local variable as they are validated only once